File: src/main/java/um/haberes/rest/model/Situacion.java
/**
 * 
 */
package um.haberes.rest.model;

import java.io.Serializable;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * @author daniel
 *
 */
@Entity
@EqualsAndHashCode(callSuper = false)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Situacion extends Auditable implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = -2895551417029745822L;

	@Id
	private Integer situacionId;

	private String nombre = "";
	private Byte interino = 0;
	private Byte ordinario = 0;
	private Byte planta = 0;
	private Byte contratado = 0;
	private Byte secundario = 0;
	
}

File: src/main/java/um/haberes/rest/kotlin/model/Situacion.kt
package um.haberes.rest.kotlin.model

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import um.haberes.rest.model.Auditable

@Entity
@Table
data class Situacion(

    @Id
    var situacionId: Int? = null,

    var nombre: String = "",
    var interino: Byte = 0,
    var ordinario: Byte = 0,
    var planta: Byte = 0,
    var contratado: Byte = 0,
    var secundario: Byte = 0

) : Auditable()
